common/helpers.py

```diff
@@ -14,13 +14,10 @@ def tables(database, *table_names):
             ...
       Where vcfs and genotypes are tables in the provided db.
     """
-    try:
-        connection = database.engine.connect()
+    with database.engine.connect() as connection:
```

Comment:
Does this answer your question, or did you mean something else?

https://docs.python.org/2/library/contextlib.html
"At the point where the generator yields, the block nested in the with statement is executed. The generator is then resumed after the block is exited"